anomaly-benchmarking/generate-datasets/script.whizzml

```diff
@@ -131,16 +153,16 @@
 ;; possible. Generates the max possible datasets (limit 40) at those
 ;; frequencies. Returns a list of datasets.
 
-(define (relative-frequency ds-id normal-rows)
+(define (relative-frequency ds-id normal-rows replicates freq-list)
   (let (ds (fetch ds-id)
         rows (ds "rows")
         ds-name (ds "name")
         my-tags (ds "tags")
         max-freq (/ rows (+ rows normal-rows)))
-    (for (freq [0.001 0.005 0.01 0.05 0.1])
+    (for (freq freq-list)
       (let (k (/ (* freq normal-rows) (- 1 freq)))
         (when (> max-freq freq)
-          (repeatedly (min 10 (floor (/ rows k))) 
+          (repeatedly (min replicates (floor (/ rows k))) 
```

Comment:
I was hoping to allow the creation of a test dataset without any anomalous points, but it makes both the other dimensions meaningless. (What's the difficulty of a point that doesn't exist? What's the variance of the empty set?) I'll put a minimum value on k (say 10) so that the variance has some good meaning.